components/Onboarding/ConnectViaWallet/ConnectViaWalletTableViewItems.tsx

```diff
@@ -131,24 +133,27 @@ export const InstalledWalletsTableView = memo(
               let walletAddress: string = "";
 
               // Specific flow for Coinbase Wallet
-              if (wallet.name === "Coinbase Wallet") {
-                const wallet = await thirdwebConnect(async () => {
-                  const coinbaseWallet = createWallet("com.coinbase.wallet", {
-                    appMetadata: config.walletConnectConfig.appMetadata,
-                    mobileConfig: {
-                      callbackURL: `https://${config.websiteDomain}/coinbase`,
-                    },
-                  });
+              if (wallet.thirdwebId === "com.coinbase.wallet") {
+                // @todo => this is a hack to remove the smart wallet key from AsyncStorage
+                // because it's not being removed by the wallet itself
+                const storageKeys = await AsyncStorage.getAllKeys();
+                const wcKeys = storageKeys.filter((k) =>
+                  k.startsWith("-Coinbase Smart Wallet:")
```

Comment:
Let's make this Coinbase smart wallet, prefix a constant somewhere and make sure it's used in all places and place of the string